### DISCUSSION TITLE
Home Assistant Core LXC upgrade

### DISCUSSION THREAD

**Discussion by fabry6210** [tteck/Proxmox#905] [Help Needed] [answered]
Hi, how to upgrade Home Assistant Core LXC ? thank's a lot.

**Comment by tteck** [comment, marked as answer]
The `***-v5.sh script` serves both to create the LXC (Proxmox Shell) and provide support for the App (LXC Console)

![Screenshot 2023-01-05 3 34 48 PM](https://user-images.githubusercontent.com/21342670/210875068-d8328bcc-3576-4bf0-9f3e-89c7d8000a96.png)

**Reply by fabry6210** [reply]
thank you,  but now, how to upgrade python 3.9.2 to 3.10 ?

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/discussions/840#discussioncomment-4441394

**Reply by fabry6210** [reply]
thank's a lot

**Reply by proddy** [reply]
like I said before, the text could be improved as it's confusing people ;-)

**Reply by tteck** [reply]
Reading is just as important , without that text doesn't matter

**Reply by fabry6210** [reply]
how to see in core LXC the dongle bluethooth USB ?

**Reply by tteck** [reply]
You mean find the location? If so, in the LXC console run `ls -l /dev/serial/by-id`
Your Home Assistant Core LXC must be created privileged to allow device passthrough.

**Reply by fabry6210** [reply]
my home assistant LXC is create priviliged, but
root@homeassistant-core1:~/.homeassistant# ls -l /dev/serial/by-id/
total 0

in proxmox
root@proxmox: lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 0a12:0001 Cambridge Silicon Radio, Ltd Bluetooth Dongle (HCI mode)
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
root@proxmox:~# ls -l /dev/bus/usb/001/003
crw-rw-r-- 1 root root 189, 2 Dec 16 19:35 /dev/bus/usb/001/003

**Reply by tteck** [reply]
Try adding
`lxc.mount.entry: /dev/bus/usb/001 dev/bus/usb/001 none bind,optional,create=dir`
to your `/etc/pve/lxc/CTID.conf` (CTID = your LXC ID #) reboot 🤞🏻

**Comment by proddy** [comment]
after the update "add arch check" (https://github.com/tteck/Proxmox/pull/981) I get

```sh
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/homeassistant-core-v5.sh)"

bash: line 421: ARCH_CHECK: command not found
‼ ERROR 127@421 Unknown failure occurred.
```

@tteck

**Reply by tteck** [reply]
I missed that one 🤷🏻‍♂️ 
It's now fixed, sorry.

**Reply by proddy** [reply]
works again. thanks

**Comment by Kirbo** [comment]
Hi, I noticed after upgrading the image, that my "Local Media" has disappeared. Could it be possible to add the local media path into the mounted volumes, so that these would also persist between upgrades?

Edit:
I just noticed that this was "Home Assistant Core LXC" and I was talking about "Home Assistant LXC". Sorry..

**Reply by tteck** [reply]
Prior to the upgrade, how was the path defined?

**Reply by Kirbo** [reply]
Let me re-phrase; the contents on the "Local media" was lost after the upgrade. I will check the path and let you know as soon as possible. Not able to do that now

**Reply by Kirbo** [reply]
Not sure does this help you in any way:
```
root@homeassistant:/var/lib/docker# find . -name 'power*.mp3'
./overlay2/4e0e0956655a31be8ac4ea53b8f78afbd65facc4b38d66cdb4f463a040585a1f/merged/media/power_off.mp3
./overlay2/4e0e0956655a31be8ac4ea53b8f78afbd65facc4b38d66cdb4f463a040585a1f/merged/media/power_on.mp3
./overlay2/4e0e0956655a31be8ac4ea53b8f78afbd65facc4b38d66cdb4f463a040585a1f/diff/media/power_off.mp3
./overlay2/4e0e0956655a31be8ac4ea53b8f78afbd65facc4b38d66cdb4f463a040585a1f/diff/media/power_on.mp3
```
https://www.home-assistant.io/more-info/local-media/setup-media/
<img width="713" alt="image" src="https://user-images.githubusercontent.com/1077974/219213838-da4ecfdb-f9e4-4826-821f-d7889ee05861.png">

**Reply by tteck** [reply]
Did you add `-v /PATH_TO_YOUR_MEDIA:/media \` to the docker run command before the upgrade?

```
docker run -d \
  --name homeassistant \
  --privileged \
  --restart unless-stopped \
  -v /var/run/docker.sock:/var/run/docker.sock \
  -v /dev:/dev \
  -v hass_config:/config \
  -v /PATH_TO_YOUR_MEDIA:/media \
  -v /etc/localtime:/etc/localtime:ro \
  -v /etc/timezone:/etc/timezone:ro \
  --net=host \
  homeassistant/home-assistant:stable
```
If you create a container to your specifications, the next time you perform an update, "runlike" will analyze the arguments in the "docker run" command line based on the currently running containers and update the existing container accordingly.

**Reply by Kirbo** [reply]
No I haven't altered the installation at all. So I should just go and modify the settings in Portainer?
Not sure did you notice, but I mentioned that I use Home Assistant Container LXC and I mistakenly commented on the Home Assistant Core LXC thread...
I installed it with this `bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/homeassistant-v5.sh)"` and upgraded by using `./update` and selecting the `Switch to Stable branch`

**Reply by Kirbo** [reply]
I take that as yes

**Reply by Kirbo** [reply]
Well, it didn't work. As soon as I updated the HA, the newly added mount in Portainer lost and now I have to add it again.

**Reply by tteck** [reply]
The reason for this is that when you choose the option "_Switch to Stable Branch_", it completely recreates everything. However, if you select "_Just Update Containers_" the program will examine the arguments in the "docker run" command line based on the currently running containers, and update them accordingly.

**Reply by tteck** [reply]
I see what you meant by your earlier message where you mentioned that you installed it using the command "`bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/homeassistant-v5.sh)`". However, version 5 of this script does not include the `./update` command, so you must have installed it using an older version of the script.

**Reply by Kirbo** [reply]
Yeah it might have been older version, can't recall anymore, I initially installed it sometime like July - September 2022 and have been updating ever since. I'll try that `Just Update Containers` then the next time I update and hope for the best!
